tests/agent_server/test_dryrun.py

```diff
@@ -313,3 +313,135 @@ async def test_dryrun_scale(resource_container, server, client, environment, age
     assert result.code == 200
 
     await agent.stop()
+
+
+@pytest.mark.asyncio(timeout=150)
+async def test_dryrun_v2(server, client, resource_container, environment):
+    """
+    Dryrun a configuration model with the v2 api, where applicable
+    """
+
+    agentmanager = server.get_slice(SLICE_AGENT_MANAGER)
+
+    agent = Agent(hostname="node1", environment=environment, agent_map={"agent1": "localhost"}, code_loader=False)
+    await agent.add_end_point_name("agent1")
+    await agent.start()
```

Comment:
Please use the `agent_factory` fixture. This one will ensure that the agent is stopped correctly, event when an exception occurs.